sources/standard/filesystem/__init__.py

```diff
@@ -0,0 +1,57 @@
+"""This source collects fsspec files."""
+import os
+from typing import Any, Dict, Optional
+
+import dlt
+from dlt.common.storages.configuration import FilesystemConfiguration
+from dlt.extract.source import TDataItems
+
+from .helpers import client_from_credentials, get_files
+from .settings import DEFAULT_CHUNK_SIZE
+
+
+@dlt.resource(
+    name="filesystem",
+    merge_key="file_url",
+    primary_key="file_url",
+    spec=FilesystemConfiguration,
+)
+def filesystem_resource(
+    credentials: FilesystemConfiguration = dlt.secrets.value,
```

Comment:
no, take a look at `FilesystemConfiguration`, what you need it:
bucket_url: str = dlt.config.value,
credentials: FileSystemCredentials = dlt.secrets.value